pkg/phases/kpack/test.go

```diff
@@ -1,16 +1,62 @@
 package kpack
 
 import (
+	"fmt"
+	"strings"
+	"time"
+
 	"github.com/flanksource/commons/console"
+	"github.com/flanksource/commons/exec"
 	"github.com/flanksource/karina/pkg/platform"
+	"github.com/flanksource/karina/pkg/types"
 	"github.com/flanksource/kommons"
 )
 
 func Test(p *platform.Platform, test *console.TestResults) {
-	if p.Kpack.IsDisabled() {
+	if p.Kpack.Disabled || (p.Kpack.ImageVersions == types.ImageVersions{}) {
 		return
 	}
 
 	client, _ := p.GetClientset()
 	kommons.TestNamespace(client, Namespace, test)
+	if p.E2E {
+		TestE2EKpack(p, test)
+	}
+}
+
+func TestE2EKpack(p *platform.Platform, test *console.TestResults) {
+	testName := "kpack-e2e-test"
+	if err := p.ApplySpecs(Namespace, "kpack-e2e.yaml"); err != nil {
+		test.Failf(testName, "Failed to apply %v", err)
+		return
+	}
+	// Wait for Builder status to be true -> wait for 5 mintues
+	for i := 0; i < 5; i++ {
+		output, ok := exec.SafeExec(fmt.Sprintf("kubectl get builder test-builder -n %s", Namespace))
```

Comment:
When calling external binaries there is a helper in `platform.GetBinary`, and also a `platform.GetKubectl` that handles creation of a local kubeconfig - there might be a kubeconfig when using kind, but our E2E tests are designed to run in production (hence the integration into karina test and not via go test) so it can't assume a kubeconfig exists when it could be dynamically generated based on the master cert.

That being said we avoid calling external binaries because it tends to be slow and fragile, the  `platform.GetByKind` function is a direct replacement for `kubectl get`

In this case it shouldn't be necessary to use GetByKind either you can use `platform.WaitForResource` that does the looping and readiness checks already